api/src/test/java/com/epam/pipeline/acl/cluster/ClusterApiServiceTest.java

```diff
@@ -0,0 +1,451 @@
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnFilteredListWithNodeInstancesForAdmin() {
+        doReturn(initSingleNodeInstanceList()).when(mockNodesManager).filterNodes(filterNodesVO);
+
+        assertThat(clusterApiService.filterNodes(filterNodesVO)).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnFilteredNodeInstanceListWhenPermissionIsGranted() {
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(initSingleNodeInstanceList()).when(mockNodesManager).filterNodes(filterNodesVO);
+
+        assertThat(clusterApiService.filterNodes(filterNodesVO)).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnFilteredListWithNodeInstanceWhichPermissionIsGranted() {
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        doReturn(authentication).when(mockAuthManager).getAuthentication();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        initAclEntity(nodeInstanceWithoutPermission, AclPermission.NO_READ);
+        doReturn(initTwoNodeInstancesList()).when(mockNodesManager).filterNodes(filterNodesVO);
+
+        assertThat(clusterApiService.filterNodes(filterNodesVO)).hasSize(1).contains(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnEmptyFilteredNodeInstanceListWhenPermissionIsNotGranted() {
+        initAclEntity(nodeInstanceWithoutPermission);
+        doReturn(initSingleNodeInstanceList()).when(mockNodesManager).filterNodes(filterNodesVO);
+
+        assertThat(clusterApiService.filterNodes(filterNodesVO)).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnNodeInstanceForAdmin() {
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+
+        assertThat(clusterApiService.getNode(nodeInstance.getName())).isEqualTo(nodeInstance);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldReturnNodeInstanceWhenPermissionIsGranted() {
+        final NodeInstance nodeInstance = NodeCreatorUtils.getNodeInstance(1L, TEST_STRING);
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        initAclEntity(nodeInstance, AclPermission.READ);
+        doReturn(nodeInstance).when(mockNodesManager).getNode(nodeInstance.getName());
+        doReturn(pipelineRun).when(mockPipelineRunManager).loadPipelineRun(eq(pipelineRun.getId()));
+
+        final NodeInstance returnedNodeInstance = clusterApiService.getNode(nodeInstance.getName());
+
+        assertThat(returnedNodeInstance).isEqualTo(nodeInstance);
+        assertThat(returnedNodeInstance.getMask()).isEqualTo(AclPermission.READ.getMask());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyAccessToNode() {
```

Comment:
Let's specify explicitly why the access is denied like `shouldDenyAccessToNodeWhenPermissionIsNotGranted`.